ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1887,39 +1908,40 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 - (BOOL)isAnswerValid:(id)answer {
     BOOL isValid = NO;
     if ([answer isKindOfClass:[NSString class]]) {
-        return [self isAnswerValidWithString:(NSString *)answer];
+        isValid = [self isAnswerValidWithString:(NSString *)answer];
     }
     return isValid;
 }
 
 - (BOOL)isAnswerValidWithString:(NSString *)text {
-    if (text.length > 0) {
-        return ([self isTextLengthValidWithString:text] && [self isEmailAddressValidWithString:text]);
+    if (text) {
+        BOOL isValid = NO;
+        if (self.regex) {
```

Comment:
We should also check for text.length > 0 here since we changed `maximumLength` from *readonly* to *readwrite*.